src/cephadm/cephadm.py

```diff
@@ -2705,6 +2718,14 @@ def get_daemon_args(ctx, fsid, daemon_type, daemon_id):
                 r += [f'--storage.tsdb.retention.size={retention_size}']
                 scheme = 'http'
                 host = get_fqdn()
+                # in case host is not an fqdn then we use the IP to
+                # avoid producing a broken web.external-url link
```

Comment:
This is a thorny issue, since technically when resolving dot-less hostnames, they should be appended to the default domains specified in the resolv.conf.